### PR TITLE
Fix  wrong settings being overwritten on apply and more important install errors

### PR DESCRIPTION
**Pull request by tekert**
All Changes:

- If `private-key` had a file path, it was overwritten by a raw key, fixed, re fix commit #6

- Overwrite only commited settings on apply button, leaving non commited settings alone. (fixes some settings being deleted, like firewall chain rules...)

- Fix -f on a glob pattern (maybe `$wgpackage` was intended?

- Fix privatekey being converted to raw key if a path was configured

- Enable checkbox now uses the interface disable node

- No internet connection can be established on `pre-config.d`; changed to `post-config.d`

- Reduce log size, it's not needed to use 1M for this. 256KB seems fine.

- Important fix on `post-config.d` script, only install wireguard if it's not installed already (previously wireguard would install on every router restart)

- `Retry` for curl online check in case we need more time to auth for internet.

- Important: if we are upgrading (a package is already present), follow the package upgrade guidelines in https://github.com/WireGuard/wireguard-vyatta-ubnt/wiki/EdgeOS-and-Unifi-Gateway#upgrade
else install the package normally.
Preserves the `config` and ` rules` of the interface if we can (important) (need testing)

- Security alert: Important: only auto install or upgrade the wizard if the user checked the boxes, this function is called from *-config.d scripts that run on router restarts, we don't want to upgrade or install the wirzard if we didn't choose to.

Tested and working on Edgerouter X v2 
pkg upgrade tested and working (checking and downloading on restart and upgrade)
pkg install not tested (need to test upgrading and install while a previous config is present)

### CODE REVIEW

**Comment by tekert** [comment]
Also the firewall chain names where beign deleted. There must be another way to disable the wg interface without deleting the node.

**Comment by tekert** [comment]
Added a better way to disabling the interface with `$cmd set $wgcfg ${interfaces[0]} disable` instead of deleting the entire node. (wich would delete so many important settings, like firewall rules and iptables scripts.

**Comment by tekert** [comment]
More important fixes:

- Fix from last commit: Enable checkbox was read incorrectly on load. (now uses the interface disable node)

- No internet connection can be established on pre-config.d; use post-config.d

- Reduce log size, it's not needed to use 1M for this. 256KB is fine.

- Important fix on post-config.d script, only install wireguard if it's not installed already (previously wireguard would install on every router restart)

- Retry for online check in case we need more time to auth for internet.

- Important: if we are upgrading (a package is already present), follow the package upgrade guidelines in https://github.com/WireGuard/wireguard-vyatta-ubnt/wiki/EdgeOS-and-Unifi-Gateway#upgrade
else install the package normally.
Preserves the config and rules of the interface if we can (important) (need testing)

- Security alert: Important: only auto install or upgrade the wizard if the user checked the boxes, this function is called from *-config.d scripts that run on router restarts, we don't want to upgrade or install the wirzard if we didn't choose to.

**Comment by tekert** [comment]
Going to test pkg install again on firmware upgrade when i have the chance, everything worked fine but on a firmware upgrade the config was not restored, fixed missing "begin", waiting on the next test on a firmware upgrade.

**Comment by vchrizz** [comment]
I merged [another pull-request](https://github.com/vchrizz/ER-wizard-WireGuard/pull/18) and now there is a conflict if I want to merge your pull-request. Could you please be so kind and have a look into this? Thank you!

**Comment by tekert** [comment]
Hey @vchrizz , sorry the long delay, I was waiting for another router update to test a real upgrade case and debug it, The idea is to restore the data in "config/config.boot" after a firmware upgrade and after wireguard package reinstall. Everything else seems to be working correctly, I've been actively using this branch on two edgerouter x. It can be merged without problems but since i can't downgrade the edgerouters I have (they are in heavy use, one of them is a public place) I wanted to make sure the config is restored.

File: wizard-run [hunk 1]
```diff
@@ -170,34 +177,76 @@ installwireguard () {
...
+        # The loaded configuration becomes the working configuration and must be committed before it becomes the active configuration.
+        # config file is saved
+        $cmd load
+        $cmd commit
+        # No need to save, config is restored from file (load) then commited to active config.
+        $cmd end
+    else
+        echo "$(date +%Y-%m-%d/%H:%M:%S.%N) [installer] Installing WireGuard for $pkgvariant ..."
+        # if we have the previous config from wireguard on config.boot (currently loaded config) then:
+        # TODO(tekert): test this
```
Comment (tekert) [inline wizard-run:205]: We have to debug this on firmware upgrade.

**Pull request merged**